test-suite/Math/NumberTheory/GaussianIntegersTests.hs

```diff
@@ -35,6 +35,7 @@ factoriseProperty1 :: Integer -> Integer -> Bool
 factoriseProperty1 x y
   =  x == 0 && y == 0
   || g == g'
+  || abs g == abs g'
```

Comment:
I guess we can safely remove check `g == g'`, because it is a specific instance of `abs g == abs g'`.